tests/test_utils/generate_cluster_desc.hpp

```diff
@@ -13,10 +13,19 @@
 namespace test_utils {
 
 inline std::string GetAbsPath(std::string path_){
-    std::filesystem::path umd_root_relative = std::filesystem::relative(std::filesystem::path(__FILE__).parent_path().parent_path().parent_path(), "../");
-    std::filesystem::path umd_root = std::filesystem::canonical(umd_root_relative);
-    std::filesystem::path abs_path = umd_root / path_;
-    return abs_path.string();
+    // Note that __FILE__ might be resolved at compile time to an absolute or relative address, depending on the compiler.
+    std::filesystem::path current_file_path = std::filesystem::path(__FILE__);
+    if (current_file_path.is_absolute()) {
+        std::filesystem::path umd_root_relative = std::filesystem::relative(std::filesystem::path(__FILE__).parent_path().parent_path().parent_path(), "../");
```

Comment:
nit: maybe you can move declarations of umd_root_relative, umd_root and abs_path before if, since all variables are used in all branches of if else. You can move abs_path.string() to the end of the function as well.